pkg/apis/flow/v1alpha1/dataweavetransformation_types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DataWeaveTransformation is the Schema for an DataWeave transformation target.
+type DataWeaveTransformation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataWeaveTransformationSpec `json:"spec"`
+	Status TargetStatus                `json:"status,omitempty"`
+}
+
+// Check the interfaces DataWeaveTransformation should be implementing.
+var (
+	_ runtime.Object     = (*DataWeaveTransformation)(nil)
+	_ kmeta.OwnerRefable = (*DataWeaveTransformation)(nil)
+	_ duckv1.KRShaped    = (*DataWeaveTransformation)(nil)
+	_ kmeta.OwnerRefable = (*DataWeaveTransformation)(nil)
```

Comment:
The interfaces which are being checked have changed since my last PR. Please see the other type files in this directory.